mne/io/egi/egimff.py

```diff
@@ -686,3 +692,199 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        all the time interval is used. Correction is applied by computing mean
+        of the baseline period and subtracting it from the data. The baseline
+        (a, b) includes both endpoints, i.e. all timepoints t such that
+        a <= t <= b.
+    %(verbose)s
+
+    Returns
+    -------
+    evoked : EvokedArray or list of EvokedArray
+        The evoked dataset(s); one EvokedArray if condition is int or str,
+        or list of EvokedArray if condition is None or list.
+
+    Raises
+    ------
+    ValueError
+        If ``fname`` has file extension other than '.mff'.
+    ValueError
+        If the MFF file specified by ``fname`` is not averaged.
+    ValueError
+        If no categories.xml file in MFF directory specified by ``fname``.
+
+    See Also
+    --------
+    Evoked, EvokedArray, create_info
+
+    Notes
+    -----
+    .. versionadded:: 0.23
+    """
+    mffpy = _import_mffpy()
+    # Confirm `fname` is a path to an MFF file
+    if not fname.endswith('.mff'):
+        raise ValueError('fname must be an MFF file with extension ".mff".')
+    # Confirm the input MFF is averaged
+    mff = mffpy.Reader(fname)
+    if mff.flavor != 'averaged':
+        raise ValueError('%s is a %s MFF file. fname must be the path to an \
+                         averaged MFF file.' % (fname, mff.flavor))
+    # Check for categories.xml file
+    if 'categories.xml' not in mff.directory.listdir():
+        raise ValueError('categories.xml not found in MFF directory. \
+                         %s may not be an averaged MFF file.' % fname)
+    if condition is None:
+        categories = mff.categories.categories
+        condition = list(categories.keys())
+    elif not isinstance(condition, list):
+        condition = [condition]
+    logger.info('Reading %d evoked datasets from %s ...'
+                % (len(condition), fname))
+    output = [_read_evoked_mff(fname, c, channel_naming=channel_naming,
+                               verbose=verbose).apply_baseline(baseline)
+              for c in condition]
+    return output if len(output) > 1 else output[0]
+
+
+def _read_evoked_mff(fname, condition, channel_naming='E%d', verbose=None):
+    """Read evoked data from MFF file."""
+    import mffpy
+    egi_info = _read_header(fname)
+    mff = mffpy.Reader(fname)
+    categories = mff.categories.categories
+
+    if isinstance(condition, str):
+        # Condition is interpreted as category name
+        category = condition
+        epoch = mff.epochs[category]
+    elif isinstance(condition, int):
+        # Condition is interpreted as epoch index
+        epoch = mff.epochs[condition]
+        category = epoch.name
+    else:
+        raise TypeError('Condition must be either int or str.')
+    assert category in categories.keys(), 'Condition "%s" not found in \
+                                           available conditions %s.'\
+                                           % (category, categories.keys())
```

Comment:
`assert` statements are skipped in optimized mode and should only be used for internal checks that should always be true (I think of them as "active comments" to remind about shapes, sizes, types, etc. that should be guaranteed)
```suggestion
    _check_option('condition', category, sorted(categories)
```
And shouldn't this go *above* the `if/else` immediately preceding it so that invalid category names are caught before `mff.epochs[category]` is attempted so that we can raise an informative error?

Reply:
Good idea to check the `condition` param first. Since `condition` can be either a `str` with the category name or an `int` with the epoch index, I will use a different check for each of these cases.